### PR TITLE
flashinfer: switch to plan API

### PR DESCRIPTION
**Pull request by danieldk**
# What does this PR do?

This change switches from the deprecated `begin_forward`/`end_forward` to `plan`.

This change doesn't switch `forward` to `run` yet, since it requires that we have access to the softmax scale and the logit softcap outside the model.

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ ] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#start-contributing-pull-requests),
      Pull Request section?
- [ ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

### CODE REVIEW

File: server/text_generation_server/layers/attention/cuda.py [hunk 1]
```diff
@@ -235,7 +235,6 @@ def attention(
             paged_kv_cache=(kv_cache.key, kv_cache.value),
             logits_soft_cap=softcap,
             sm_scale=softmax_scale,
-            window_left=window_size_left,
```
Comment (Narsil) [inline server/text_generation_server/layers/attention/cuda.py:238]: Why is this gone ?

Comment (danieldk) [reply server/text_generation_server/layers/attention/cuda.py:238]: It is already set by the context manager:

https://github.com/huggingface/text-generation-inference/blob/a0a0316ae841ea6747fbae6c4990bb9e15c079ae/server/text_generation_server/layers/attention/flashinfer.py#L98
https://github.com/huggingface/text-generation-inference/blob/a0a0316ae841ea6747fbae6c4990bb9e15c079ae/server/text_generation_server/layers/attention/flashinfer.py#L212

This is one of the arguments that is gone in the new `run()` API.

**Review by Narsil — APPROVED** [review]
